analyzers/tests/SonarAnalyzer.UnitTest/TestCases/InvocationResolvesToOverrideWithParams.TopLevelStatements.cs

```diff
@@ -0,0 +1,71 @@
+﻿Repro5430.SomeClass x;
+x = new ("1");              // Compliant, can't see non-param overload
+x = new ("1", "s1");        // Compliant, can't see non-param overload
+x = new (null, "s1");       // Compliant, can't see non-param overload
+x = new ("1", "s1", "s2");  // Compliant, can't see non-param overload
+x = new (null, "s1", "s2"); // Compliant, can't see non-param overload
+
+x.PrivateOverload("42");                           // Compliant
+x.PrivateOverload("42", "s1");                     // Compliant
+x.PrivateOverload(null, "s1");                     // Compliant
+x.PrivateOverload(null, new[] { "s2" });           // Compliant
+x.PrivateOverload("42", "s1", "s2");               // Compliant
+
+x.ProtectedOverload("s1");                         // Compliant
+x.ProtectedOverload("s1", "s2");                   // Compliant
+x.ProtectedOverload(null, "s2");                   // Compliant
+x.ProtectedOverload(null, new[] { "s2" });         // Compliant
+x.ProtectedOverload("42", "s1", "s2");             // Compliant
+
+x.PrivateProtectedOverload("s1");                  // Compliant
+x.PrivateProtectedOverload("s1", "s2");            // Compliant
+x.PrivateProtectedOverload(null, "s2");            // Compliant
+x.PrivateProtectedOverload(null, new[] { "s2" });  // Compliant
+x.PrivateProtectedOverload("42", "s1", "s2");      // Compliant
+
+x.ProtectedInternalOverload("s1");                 // Compliant
+x.ProtectedInternalOverload("s1", "s2");           // Noncompliant
+x.ProtectedInternalOverload(null, "s2");           // Noncompliant
+x.ProtectedInternalOverload(null, new[] { "s2" }); // Compliant
+x.ProtectedInternalOverload("42", "s1", "s2");     // Compliant
+
+x.InternalOverload("s1");                          // Compliant
+x.InternalOverload("s1", "s2");                    // Noncompliant
+x.InternalOverload(null, "s2");                    // Noncompliant
+x.InternalOverload(null, new[] { "s2" });          // Compliant
+x.InternalOverload("42", "s1", "s2");              // Compliant
+
+// Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/5430
+namespace Repro5430
+{
```

Comment:
I would remove it from here. If any code change will fail, it should fail on the main file as well and there already is the explanation and link.

If you would insist on having `Repro_5430` here, it could be name of the class, without a namespace.
```suggestion
```